### ISSUE TITLE
Upgrade PHPMailer to 5.2.16. Use XOAUTH2 classs for secure GMail.

### ISSUE THREAD

**Issue by chris001** [salesagility/SuiteCRM#2512]
#### Issue

1. PHPMailer 5.2.13 is outdated.  Should be upgraded to 5.2.16.
2. PHPMailer comes with the XOAUTH2 class, which recommended by Google as the login method for applications to securely authenticate to GMail.  The purpose of XOAUTH2 login is to prevent theft of your Google account, much greater protection from email account hacking, etc.  Google disables by default normal password based login because allowing applications to perform password login to email is terribly insecure.  Getting XOAUTH2 into SuiteCRM would eliminate the constant problems users are experiencing in trying to connect suitecrm to one of the biggeset email providers, gmail.
#### Expected Behavior

Send email thru GMail with no errors.
#### Actual Behavior

Error LOGIN-PLAIN unsupported by default, use XOAUTH2, or enable "Less secure apps" which is easily hacked.
#### Possible Fix

Here is an example of using PHPMailer to login to gmail with XOAUTH2, and sending an email.

https://github.com/PHPMailer/PHPMailer/blob/master/examples/gmail_xoauth.phps
#### Steps to Reproduce

1.
2.
3.
4.
#### Your Environment

- SuiteCRM Version used: 7.7.6
- Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)): Firefox 50.0b10
- Environment name and version (e.g. MySQL, PHP 7): PHP  5.6, Mysql 5.6
- Operating System and version (e.g Ubuntu 16.04): Debian 8

**Comment by chris001** [comment]
See #2831 
PHPMailer should be upgraded urgently to 5.2.19 due to security issue, remote code execution:
https://legalhackers.com/advisories/PHPMailer-Exploit-Remote-Code-Exec-CVE-2016-10033-Vuln.html

* This issue makes it more urgent for SuiteCRM to start to actively use PSR composer/packagist.org.  All user/customer installations of SuiteCRM could be easily configured so that composer would by default download and upgrade both the CRM application itself and the 3rd party libraries such as PHPmailer automatically when a new important bug fix or security fix version is released.

**Comment by chris001** [comment]
PR #2840 should fix this by means of `composer update`, running on a daily `cron` job.

**Comment by horus68** [comment]
@chris001 see related (PHP mailer update) https://github.com/salesagility/SuiteCRM/pull/3340

**Comment by chris001** [comment]
Thanks @horus68 . Automatic security updates of 3rd party libraries is adding to reasons to merge #3237 now.

**Comment by samus-aran** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Upgrade PHPMailer to 5.2.16. Use XOAUTH2 classs for secure GMail.](https://trello.com/c/OaLpSyFH/59-upgrade-phpmailer-to-5-2-16-use-xoauth2-classs-for-secure-gmail)

**Comment by samus-aran** [comment]
We have now set up a new home for suggestions at Trello.  All github issues that were labeled 'suggestion' have been moved and will be closed.  Certain ones will be progressed within the new Suggestion Box and may be re-opened.

Announcement of moving Suggestions:
https://suitecrm.com/forum/suggestion-box/13691-moving-suggestions-from-github-to

New SuiteCRM Suggestion Box
https://trello.com/b/Ht7LbMqw/suitecrm-suggestion-box
